symengine/cwrapper.cpp

```diff
@@ -1832,6 +1834,21 @@ CWRAPPER_OUTPUT_TYPE basic_as_numer_denom(basic numer, basic denom,
     CWRAPPER_END
 }
 
+CWRAPPER_OUTPUT_TYPE basic_as_two_terms(basic term1, basic term2, const basic s)
```

Comment:
How about using two methods `basic_add_as_two_terms` and `basic_mul_as_two_terms`? This isn't a general method that applies to all basic right?

Reply:
Well I was going by sympy standards where we get an error when we use `as_two_terms` on any expression which is not an `Add` or `Mul` object. But we get an output when the same function is called on either `Add` or `Mul`
```
>>> x = Symbol('x')
>>> y = Symbol("y")
>>> (x**y).as_two_terms()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Pow' object has no attribute 'as_two_terms'
>>> (x + y).as_two_terms()
(x, y)
>>> (x * y).as_two_terms()
(x, y)
```

I kinda like this approach more than having two separate `as_two_terms` methods (even if we split, we would take the same approach I reckon , for eg  for `basic_mul_as_two_terms` first we assert the `Mul` type and then call the `as_two_terms`)

Technically (as also seen in my use case), `as_two_terms` is seen being used only after there is a type check (both or just `Mul` or just `Add`) in sympy's code base. 
```
    elif e.is_Mul or e.is_Add:
        i, d = e.as_independent(x)  # throw away x-independent terms
        if d.func != e.func:
            s, expr = mrv(d, x)
            return s, e.func(i, expr)
        a, b = d.as_two_terms()     
```
So we don't really encounter a use case in sympy's internal's atleast where we have to split the two. My thinking tells me we can maybe stick to one for now and if we see a use case later one split the two. Does that sound convincing to you ?

Reply:
I do get your point though, the name `basic_as_two_terms` might be deceiving if the function won't be applicable for all basics. We can maybe change the name and mentiond (add & mul) in the name.

Reply:
I would split it into two methods, one for mul, one for add. The if statement inside it feels like the wrong abstraction. Just expose the methods in a clean manner, we can build the exact abstractions we need in Python/LPython itself on top of this.